### PR TITLE
Fixing bug where templates may get mapped before loaded from their files

### PR DESCRIPTION
**Pull request by Benjin**
Some script templates were getting added to a map before the main controller had a chance to initialize them.  This PR merges the mapping logic into the initialization logic to ensure order, checks initialization when when getting the maps, and adds tests to verify file loading and negative cases.

### CODE REVIEW

File: extensions/sql-database-projects/src/templates/templates.ts [hunk 1]
```diff
@@ -4,27 +4,85 @@
...
+export const storedProcedure: string = 'storedProcedure';
+export const folder: string = 'folder';
+
+// Object maps
+
+let scriptTypeMap: Record<string, ProjectScriptType> = {};
+
+export function projectScriptTypeMap(): Record<string, ProjectScriptType> {
+	if (Object.keys(scriptTypeMap).length === 0) {
+		throw new Error('Templates must be loaded from file before attempting to use.');
```
Comment (llali) [inline extensions/sql-database-projects/src/templates/templates.ts:26]: add to localizes texts

File: extensions/sql-database-projects/src/templates/templates.ts [hunk 2]
```diff
@@ -4,27 +4,85 @@
...
 export async function loadTemplates(templateFolderPath: string) {
+
 	newSqlProjectTemplate = await loadTemplate(templateFolderPath, 'newSqlProjectTemplate.xml');
 
-	newSqlScriptTemplate = await loadTemplate(templateFolderPath, 'newTsqlScriptTemplate.sql');
-	newSqlTableTemplate = await loadTemplate(templateFolderPath, 'newTsqlTableTemplate.sql');
-	newSqlViewTemplate = await loadTemplate(templateFolderPath, 'newTsqlViewTemplate.sql');
-	newSqlStoredProcedureTemplate = await loadTemplate(templateFolderPath, 'newTsqlStoredProcedureTemplate.sql');
+	await loadObjectTypeInfo(script, constants.scriptFriendlyName, templateFolderPath, 'newTsqlScriptTemplate.sql');
+	await loadObjectTypeInfo(table, constants.tableFriendlyName, templateFolderPath, 'newTsqlTableTemplate.sql');
```
Comment (llali) [inline extensions/sql-database-projects/src/templates/templates.ts:47]: is it possible to load all these as parallel?

Comment (Benjin) [reply extensions/sql-database-projects/src/templates/templates.ts:47]: Good idea.  Yeah, they can be.

File: extensions/sql-database-projects/src/templates/templates.ts (hunk 1 above)
Comment (Benjin) [reply extensions/sql-database-projects/src/templates/templates.ts:26]: This error is more of an assert, and not meant to be user-facing.  Do we still localize those?

**Pull request merged**